core/base/inc/Bytes.h

```diff
@@ -67,9 +67,10 @@ inline void tobuf(char *&buf, UChar_t x)
    *buf++ = x;
 }
 
-inline void tobuf(char *&buf, UShort_t x)
+inline void tobuf(char *&buf, UShort_t x, Bool_t buffBigEndian = kTRUE)
 {
 #ifdef R__BYTESWAP
+if(buffBigEndian) {
```

Comment:
What is the 'cost' of this added if statement (it is in the inner most part of the processing)?